### ISSUE TITLE
Missing functions in models.py

### ISSUE THREAD

**Issue by roman-studer** [shiming-chen/HSVA#2]
The following functions seem to be missing in `models.py`:

- `models.discriminator_template()`
- `models.decoder_z()`
- `models.domain_classifier()`
- `models.domain_discriminator()`

Can you make these available? 

Kind Regards

**Comment by LZhongYue** [comment]
Sir, have you solved this problem? @roman-studer

**Comment by shiming-chen** [comment]
Hi, @roman-studer @LZhongYue 
Thanks for your attention on our work. We have added the the related functions in `model.py`.
Best!

Shiming

**Comment by LZhongYue** [comment]
The HSVA.py of the file is missing train_vae() function, I originally wanted to take it from the CADA-VAE project, but the version of torch he wrote is too old and will report an error, you can update it when you are free
![image](https://user-images.githubusercontent.com/66252224/233300946-72613b2a-ba25-420a-9691-723424e01e59.png)

**Comment by LZhongYue** [comment]
Thank you Dr. Chen, I have successfully run HSVA， I hope one day I can also post the top conference☀@shiming-chen

**Comment by roman-studer** [comment]
Thank you very much for the reply and the changes. I have also been able to run HSVA.

FYI, I had to change `self.discriminator[datatype] = models.discriminator_template(dim, 1, self.hidden_size_rule[datatype], self.device)` in HSVA.py to `self.discriminator[datatype] = models.domain_discriminator(dim, 1, self.hidden_size_rule[datatype], self.device)` and am assuming that `models.domain_discriminator` functions the same as `models.discriminator_template`.

Kind regards
